cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/utils/XXHash32DigestAlgorithm.java

```diff
@@ -61,7 +61,10 @@ public Digest calculateFileDigest(Path path) throws IOException
             {
                 hasher.update(buffer, 0, len);
             }
-            return new XXHash32Digest(Long.toHexString(hasher.getValue()), SEED);
+            // lz4 library doesn't mask the hash value, so we need to mask it to
+            // prevent forwarding the negative sign bit when converting to a long value
+            long hash = hasher.getValue() & 0xffffffffL;
```

Comment:
The comment is not true. 
There is no mask required for lz4's implementation since it returns `int`. Masking is required because our code casts `int` to `long`, and the sign bit cannot be carried over. 
Alternatively, you can just call this w/o masking.

```java
Integer.toHexString(hasher.getValue())
```

And the proof.
```
jshell> int i = -1;
i ==> -1

jshell> Integer.toHexString(i);
$2 ==> "ffffffff"

jshell> long l = i & 0xffffffffL;
l ==> 4294967295

jshell> Long.toHexString(l);
$4 ==> "ffffffff"
```

Reply:
that makes sense. Thanks for the clarification. I've pushed a commit to address this.